### ISSUE TITLE
[RFE] Add functionality or Option to Set/Modify/Remove the priority according to the job schedule or running task

### ISSUE THREAD

**Issue by sjha4** [pulp/pulpcore#2850]
**Is your feature request related to a problem? Please describe.**
Is it possible to add functionality, where we can set the High/Low priority of tasks like repo sync/capsule sync, for that some pulp workers are reserved for "other activities than capsule sync"

**Describe the solution you'd like**
When there are multiple LE's or Multiple CV's and when we have more number connected external capsules start syncing in this scenario, there are chances to have the performance issue or it takes much time,

So in that case, if we can set the low/High priority for a capsule sync task,
Satellite/Capsule will reserve a few workers and slowly/steadily side by side it will send/receive data.

Whereas we will have resources to perform another task as well or there will not be resource issues if we can divide those workers between 2 different tasks simultaneously.

**Comment by bmbouter** [comment]
Yes this is technically possible, but I'm very hesitant to add "service classes" to the tasking system, or some sort of priority queue. I'm reading that the issue is that work is waiting in the queue longer than is desired. I'm not reading that we actually have different classes of service. I think the actual issue is there aren't enough workers for the workload being run.

Typically Sync operations are the long running ones. So prioritizing long-running operations ahead of a bunch of short-running operations increases the average waiting time of items in the queue, which is the very thing we're trying to decrease. If we're talking prioritizing one sync over another, do uses have a preference for one over another?

Classes of service are tough to manager, which is my primary concern. There are a few things I worry about with that:

* Starvation. It's very possible that the high prio work will endlessly consume the workers causing some work in Pulp to just never be handled

* Fairness. The First come first serve queueing policy is the fairest queueing policy of them all, mathematically resulting in the minimized average waiting time in the queue and minimized average variance of all tasks in the system. Navigating this in multi-user pulp installations for example would be very difficult.

* The service classes could conflict with the locking requirements. For example, say users could submit with their jobs a priority number, with higher priorities being handled first. Users submits a sync for repo `foo` with prio 1.0 and then submits a publish for repo `foo` with prio 2.0. Assume both wait in the queue because no available worker is ready. The locking system says that the submission order requires the sync to occur before the publish (which for obvious reasons is right), but now the service classes says the order should be publish, then sync. These features are now conflicting.

My goal is to help. What about instead we could identify Pulp queue wait times are getting too long? Then users could choose to either a) reduce the workloads or b) increase their number of workers. Would that possibly be helpful?
